tools/global_bss_check.py

```diff
@@ -0,0 +1,216 @@
+
+    # If it's bad or missing, don't need to do anything special.
+    # If it's good, check for if it's in a file with bad or missing stuff, and check if print all is on. If none of these, print it.
+    
+    for symbol in comparedDict:
+        symbolInfo = comparedDict[symbol]
+        symbolGood = colorama.Fore.RED + "BAD" + colorama.Fore.RESET
+        if type(symbolInfo.diff) != int:
+            symbolGood = colorama.Fore.YELLOW + "MISSING" + colorama.Fore.RESET
+            print(f"{symbol},{symbolInfo.buildAddress:X},{symbolInfo.buildFile},{symbolInfo.expectedAddress:X},{symbolInfo.expectedFile},{symbolInfo.diff},{symbolGood}")
+            continue
+
+        if symbolInfo.diff == 0:
+            symbolGood = colorama.Fore.GREEN + "GOOD" + colorama.Fore.RESET
+            if (not symbolInfo.buildFile in badFiles and not symbolInfo.expectedFile in badFiles) and (not symbolInfo.buildFile in badFiles and not symbolInfo.expectedFile in badFiles) and not printAll:
+                continue
+        
+        if symbolInfo.buildFile != symbolInfo.expectedFile:
+            symbolGood += colorama.Fore.CYAN + " MOVED" + colorama.Fore.RESET
+        print(f"{symbol},{symbolInfo.buildAddress:X},{symbolInfo.buildFile},{symbolInfo.expectedAddress:X},{symbolInfo.expectedFile},{symbolInfo.diff:X},{symbolGood}")
+
+
+def main():
+    description = "Check that globally visible bss has not been reordered."
+    # TODO
+    epilog = """\
+    N.B. Since this script reads the map files, it can only see globally visible bss; in-function static bss must be examined with other tools.
+    """
+
+    parser = argparse.ArgumentParser(description=description, epilog=epilog, formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("mapFile", help="Path to a map file.")
+    parser.add_argument("mapFileExpected", help="Path to the expected map file. Optional, default is 'expected/mapFile'.", nargs="?", default="")
+    parser.add_argument("-a", "--print-all", help="Print all bss, not just non-matching.", action="store_true")
+    parser.add_argument("-n", "--no-fun-allowed", help="Remove amusing messages.", action="store_true")
+    args = parser.parse_args()
+
+    if args.mapFileExpected == "":
+        args.mapFileExpected = os.path.join("expected", args.mapFile)
+
+    badFiles, missingFiles, comparedDict = compareMapFiles(args.mapFile, args.mapFileExpected)
+    printCsv(badFiles, missingFiles, comparedDict, args.print_all)
+
+    if len(badFiles) + len(missingFiles) != 0:
+        print("", file=os.sys.stderr)
+
+    if len(badFiles) != 0:
+        print(colorama.Fore.RED + "  BAD" + colorama.Style.RESET_ALL)
+
+        for file in badFiles:
+            print(f"bss reordering in {file}", file=os.sys.stderr)
+        print("", file=os.sys.stderr)
+        
+        if not args.no_fun_allowed:
+            print(colorama.Fore.LIGHTWHITE_EX +
+            "  BSS is REORDERED!!\n"
+            "  Oh! MY GOD!!" 
```

Comment:
If anyone was wondering, this message comes from a debugging screen the devs built in for when the RCP gets hung. I 100% approve of its inclusion in this PR 💯 

![rcp](https://user-images.githubusercontent.com/31797144/135629228-051f439e-ea22-4612-976c-a62fcec5b8c8.png)